src/main/java/com/mastercard/developer/encryption/CryptoProvider.java

```diff
@@ -0,0 +1,85 @@
+package com.mastercard.developer.encryption;
+
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.Option;
+import com.jayway.jsonpath.spi.json.JsonProvider;
+import com.mastercard.developer.json.JsonEngine;
+
+import java.util.Collection;
+import java.util.Collections;
+
+abstract class CryptoProvider {
```

Comment:
I wouldn't use inheritance here, as it's mainly a way of sharing stuff. It would be better to use delegation and give to `FieldLevelEncryption` and `JweEncryption` an instance of `CryptoProvider` (of course with a better classname since we are talking about parsing/handling a JSON document, not "crypto").